packages/@aws-cdk/aws-nodejs-aspect/lib/nodejs-aspect.ts

```diff
@@ -0,0 +1,137 @@
+    const targetNode = functionNode?.node.children.find((child) => child instanceof lambda.CfnFunction) as lambda.CfnFunction;
+    targetNode.runtime = this.targetRuntime;
+  }
+
+  private handleCustomResourceProvider(provider: cdk.CustomResourceProviderBase): void {
+    const node = provider as cdk.CustomResourceProviderBase;
+    // only replace for nodejs case
+    if (this.isValidRuntime(node.runtime)) {
+      const targetnode = this.getChildFunctionNode(node);
+      targetnode.addPropertyOverride('Runtime', this.targetRuntime);
+    }
+  }
+
+  private handleProvider(provider: Provider): void {
+    const functionNodes = provider.node.children.filter((child) => child instanceof lambda.Function);
+    for ( var node of functionNodes) {
+      const targetNode = this.getChildFunctionNode(node);
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        targetNode.addPropertyOverride('Runtime', this.targetRuntime);
+      }
+
+      // onEvent Handlers
+      const onEventHandler = provider.onEventHandler as lambda.Function;
+      const onEventHandlerRuntime = this.getChildFunctionNode(onEventHandler);
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        onEventHandlerRuntime.addPropertyOverride('Runtime', this.targetRuntime);
+      }
+
+      //isComplete Handlers
+      // Handlers
+      const isCompleteHandler = provider.isCompleteHandler as lambda.Function;
+      const isCompleteHandlerRuntime = this.getChildFunctionNode(isCompleteHandler);
+      if (this.isValidRuntime(this.getRuntimeProperty(targetNode))) {
+        isCompleteHandlerRuntime.addPropertyOverride('Runtime', this.targetRuntime);
+      }
+    }
+  }
+
+  private handleReceiptRuleSet(ruleSet: ReceiptRuleSet): void {
+    const functionnode = ruleSet.node.findChild('DropSpam');
+    const ses = functionnode.node.findChild('Function') as lambda.CfnFunction;
+    if (ses.runtime && this.isValidRuntime(ses.runtime)) {
+      ses.addPropertyOverride('Runtime', this.targetRuntime);
+    }
+  }
+
+  /**
+ *Runtime Validation
+ *
+ */
+  private isValidRuntime(runtime: string) {
+    const family = this.getRuntimeFamily(runtime);
+    return family === lambda.RuntimeFamily.NODEJS;
+  }
+
+  private getRuntimeProperty(node: cdk.CfnResource) {
+    return (node.getResourceProperty('runtime') || node.getResourceProperty('Runtime')) as string;
+  }
+
+  private getChildFunctionNode(node: Construct) : cdk.CfnResource {
+    const childNode = node.node.children.find((child) => cdk.CfnResource.isCfnResource(child) && child.cfnResourceType === 'AWS::Lambda::Function') as cdk.CfnResource;
+    return childNode;
+  }
+
+  private getRuntimeFamily(runtime: string) {
+    switch (runtime) {
+      case 'nodejs18.x':
+        return lambda.Runtime.NODEJS_18_X.family;
+      default:
+        return 'Unsupported';
+    }
+  }
+
+}
+
+/**
+ * RuntimeAspect
+ */
+export class RuntimeAspect extends RuntimeAspectsBase {
```

Comment:
Do we really want to expose this? Additionally, if I'm using this I'm not clear what `key` is. Also, `string` is extremely loose. I could pass anything as a string. It seems like we should be building an enum-like class on `RuntimeAspect` which is extensible. Introducing `NodeJsAspect` means we would need a new class for every single runtime. I realize we may never introduce new runtimes, but we should make this extensible. How about something like this:
```ts
export class RuntimeAspect extends RuntimeAspectsBase {
	public static nodeJs20() {
		// return class implementation
	}
}
```

This may be better to put everything on `RuntimeAspectsBase` though since we would just be returning the implementation and another class is probably extra unnecessary code.

Reply:
updated in latest commit for your review